### PR TITLE
Io.snapcraft.privileged desktop launcher

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: data/dbus/io.snapcraft.PrivilegedDesktopLauncher.service.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[D-BUS Service]
+Name=io.snapcraft.PrivilegedDesktopLauncher
+Exec=@bindir@/snap userd
+AssumedAppArmorLabel=unconfined
```
Comment (jhenstridge) [inline data/dbus/io.snapcraft.PrivilegedDesktopLauncher.service.in:4]: I would strongly recommend against adding another bus name.  Clients should be able to access the new APIs by making calls to the existing `io.snapcraft.Launcher` bus name.

The existing code inside userd conflates bus names with interface names, but I wouldn't exacerbate it by adding more service activation files.

Comment (jhenstridge) [reply data/dbus/io.snapcraft.PrivilegedDesktopLauncher.service.in:4]: With #9370 merged, this file can be removed.

File: usersession/userd/privileged_desktop_launcher.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+	</method>
+</interface>`
+
+// PrivilegedDesktopLauncher implements the 'io.snapcraft.PrivilegedDesktopLauncher' DBus interface.
+type PrivilegedDesktopLauncher struct {
+	conn *dbus.Conn
+}
+
+// Name returns the name of the interface this object implements
+func (s *PrivilegedDesktopLauncher) Name() string {
```
Comment (jhenstridge) [inline usersession/userd/privileged_desktop_launcher.go:62]: This method should be renamed to `Interface`

File: usersession/userd/privileged_desktop_launcher.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	return "io.snapcraft.PrivilegedDesktopLauncher"
+}
+
+// BasePath returns the base path of the object
+func (s *PrivilegedDesktopLauncher) BasePath() dbus.ObjectPath {
```
Comment (jhenstridge) [inline usersession/userd/privileged_desktop_launcher.go:67]: ... and this one to `ObjectPath`

**Review by jhenstridge — CHANGES_REQUESTED** [review]
A few small changes are needed to work after #9370 was landed, but overall this looks good.

File: data/dbus/io.snapcraft.PrivilegedDesktopLauncher.service.in (hunk 1 above)
Comment (AlanGriffiths) [reply data/dbus/io.snapcraft.PrivilegedDesktopLauncher.service.in]: Thanks @jhenstridge that works nicely

**Pull request merged**